rand_core/src/impls.rs

```diff
@@ -407,20 +400,42 @@ impl<R: BlockRngCore> BlockRng64<R> {
         }
     }
 
+    /// Return a reference the wrapped `BlockRngCore`.
+    pub fn inner(&self) -> &R {
+        &self.core
+    }
+
     /// Return a mutable reference the wrapped `BlockRngCore`.
-    pub fn inner(&mut self) -> &mut R {
+    pub fn inner_mut(&mut self) -> &mut R {
```

Comment:
Since we must expose both `inner` and `inner_mut` and we will always have exactly one "core", there doesn't seem to be any advantage over just making `core` a `pub` field (which would be nicer syntactically). Since we have to make a breaking change here anyway (and probably no other users will be affected yet), shall we just do this?

For the other fields there is *some* advantage in keeping them private, but not really for `core` I think.

Reply:
I don't think it matters much one way or the other. It feels a little strange to me to have one public field, but the rest private.

Reply:
It works reasonably well IMO — typical usage is like `ChaChaRng(BlockRng { core: ChaChaCore, .. })`; i.e. the outer type is free to access the inner type however is deemed fit, whereas the other fields are not quite so straightforward.